File: recursividade/python/mesclar-listas.py
def merge_lists(a, b):
    if not a:
        return b
    if not b:
        return a

    head1 = a[0]
    tail1 = a[1:]

    head2 = b[0]
    tail2 = b[1:]

    return [head1, head2] + merge_lists(tail1, tail2)


result1 = merge_lists([10, 20, 30], [5, 8, 7])
print(result1)

result2 = merge_lists(["ana", "maria"], ["joao", "bob", "alex", "leo"])
print(result2)

File: recursividade/javascript/mesclar-listas.js
function mergeLists(a, b) {
  if (a.length === 0) {
    return b;
  }
  if (b.length === 0) {
    return a;
  }

  const head1 = a[0];
  const tail1 = a.slice(1);

  const head2 = b[0];
  const tail2 = b.slice(1);

  return [head1, head2].concat(mergeLists(tail1, tail2));
}

const result1 = mergeLists([10, 20, 30], [5, 8, 7]);
console.log(result1);

const result2 = mergeLists(["ana", "maria"], ["joao", "bob", "alex", "leo"]);
console.log(result2);
